src/vcpkg/postbuildlint.cpp

```diff
@@ -1025,6 +1027,101 @@ namespace vcpkg::PostBuildLint
         return LintStatus::SUCCESS;
     }
 
+    static LintStatus check_no_absolute_paths_in(const Filesystem& fs, const Path& dir, Span<Path> absolute_paths)
+    {
+        static constexpr StringLiteral extensions[] = {"h", "hpp", "hxx", "py", "sh", "cmake", "pc", "cfg", "conf"};
+        std::vector<std::pair<Path, std::string>> files_and_contents;
+        for (auto& path : fs.get_regular_files_recursive(dir, IgnoreErrors{}))
+        {
+            if (path.extension().empty())
+            {
+                auto contents = fs.read_contents(path, VCPKG_LINE_INFO);
+                if (Strings::starts_with(contents, "#!"))
+                {
+                    files_and_contents.emplace_back(std::move(path), std::move(contents));
+                }
+            }
+            else if (Util::contains(extensions, path.extension().substr(1 /* ignore dot */)))
+            {
+                auto contents = fs.read_contents(path, VCPKG_LINE_INFO);
+                files_and_contents.emplace_back(std::move(path), std::move(contents));
+            }
+        }
+        std::vector<std::string> string_paths;
+        for (const auto& path : absolute_paths)
+        {
+#if defined(_WIN32)
+            auto path_preferred = path;
+            path_preferred.make_preferred();
+            string_paths.push_back(path_preferred.generic_u8string());
+            string_paths.push_back(std::move(path).native());
+#else
+            string_paths.push_back(path.native());
+#endif
+        }
+
+        std::string result;
+        for (const auto& path_and_contents : files_and_contents)
+        {
+            const auto extension = path_and_contents.first.extension().substr(1 /* ignore dot */);
+            const bool is_header = extension == "h" || extension == "hpp" || extension == "hxx";
+            if (Util::any_of(string_paths, [&path_and_contents, extension, is_header](const std::string& path) {
```

Comment:
```suggestion
            if (Util::any_of(string_paths, [&path_and_contents, extension, is_header](StringView path) {
```

Would be nice if we could use `StringView` here.

Reply:
That won't do anything, the input is already a `vector<std::string>`.